### PR TITLE
docs: update protocol assumptions

### PR DESCRIPTION
**Pull request by smol-ninja**
Update protocol assumption post Cantina finding. Also added assumption about tokens with admin powers as pointed by @razgraf on Slack.

Closes https://github.com/sablier-labs/flow/issues/310.

@andreivladbrg I have replaced the "delay" keyword with "precision errors" and "loss" as thats what Cantina auditors used in their submissions. I think these terms are easier to follow compared to "delay" keywords.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -44,18 +44,30 @@ Vulnerabilities contingent upon the occurrence of any of the following also are
...
+- The total supply of any ERC-20 token remains below $(2^{128} - 1)$, i.e., `type(uint128).max`.
+
 - The `transfer` and `transferFrom` methods of any ERC-20 token strictly reduce the sender's balance by the transfer
   amount and increase the recipient's balance by the same amount. In other words, tokens that charge fees on transfers
   are not supported.
+
 - An address' ERC-20 balance can only change as a result of a `transfer` call by the sender or a `transferFrom` call by
   an approved address. This excludes rebase tokens and interest-bearing tokens.
+
+- The admin of the ERC-20 contract does not burn tokens that are held in the Flow contract.
```
Comment (PaulRBerg) [inline SECURITY.md:56]: This appears to be a special case of the previous assumption, so I will incorporate it there.

**Review by PaulRBerg — APPROVED** [review]
Thanks, Shub. I don't like how it looks with new lines in between the assumptions. Would you mind if I removed them?

**Comment by smol-ninja** [comment]
Yeah go ahead @PaulRBerg

**Comment by PaulRBerg** [comment]
Thanks, good to merge now.

We might want to add the same assumption about the admin in the Lockup security document.

**Comment by smol-ninja** [comment]
Thank you. 

Yes there is a separate issue to track that: https://github.com/sablier-labs/v2-core/issues/1065.

**Pull request merged**